content/docs/pipelines/pipelines-overview.md

```diff
@@ -155,6 +155,54 @@ Receiver operating characteristics (ROC) curve:
   alt="ROC"
   class="mt-3 mb-3 p-3 border border-info rounded">
 
+## Architectural overview
+
+<img src="/docs/images/pipelines-architecture.png" 
+  alt="Pipelines architectural diagram"
+  class="mt-3 mb-3 p-3 border border-info rounded">
+
+At a high level, the execution of a pipeline proceeds as follows: 
+
+* **Python SDK**: You create components or specify a pipeline using the Kubeflow
+  Pipelines domain-specific language 
+  ([DSL](https://github.com/kubeflow/pipelines/tree/master/sdk/python/kfp/dsl)).
+* **DSL compiler**: The
+  [DSL compiler](https://github.com/kubeflow/pipelines/tree/master/sdk/python/kfp/compiler)
+  transforms your pipeline's Python code into a static configuration (YAML).
+* **Pipeline Service**: You call the Pipeline Service to create a
+  pipeline run from the static configuration. 
+* **Kubernetes resources**: The Pipeline Service calls the Kubernetes API
+  server to create the necessary Kubernetes resources
+  ([CRDs](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/))
+  to run the pipeline.   
+* **Orchestration controllers**: A set of orchestration controllers
+  execute the containers needed to complete the pipeline execution specified
+  by the Kubernetes resources
+  ([CRDs](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/)).
+  The containers execute within Kubernetes Pods on virtual machines. An
+  example controller is the **[Argo
+  Workflow**](https://github.com/argoproj/argo) controller, which
+  orchestrates task-driven workflows. 
+* **Artifact storage**: The Pods store two kinds of data: they call a Data
+  Artifact Storage Service (Minio Server, Google Cloud Storage, etc. to store
+  data artifacts. They call a Metrics Artifact Storage Service (GCP Stackdriver,
```

Comment:
Currently metrics are stored as part of the metadata, instead of gcp stackdriver. @hongye-sun might be the best person to review this item.

Reply:
I've made some adjustments. @hongye-sun PTAL.

Reply:
We currently only store single scalar metric on the run's metadata. For large metrics data like tensorboad metrics, we stored them in artifact store like GCS. Those are two types of metrics. The metrics on metadata are usually aggregated metrics which will be used for sorting and filtering purpose. The large metrics like time-series metrics are usually used to investigate individual run's performance and debugging purpose.

Reply:
Thanks. I've updated the text. PTAL
https://deploy-preview-614--competent-brattain-de2d6d.netlify.com/docs/pipelines/pipelines-overview/#architectural-overview